modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1407,6 +1435,49 @@ void ONNXImporter::parseMul(LayerParams& layerParams, const opencv_onnx::NodePro
             //Replace input to Power
             node_proto.set_input(1, powerParams.name);
         }
+
+        const MatShape& broadShape = outShapes[node_proto.input(1)];
+        const MatShape& outShape = outShapes[node_proto.input(0)];
+        const size_t outShapeSize = outShape.size();
+        const size_t diff = outShapeSize - broadShape.size();
+
+        size_t axis = diff;
+        for (; axis < broadShape.size() && broadShape[axis - diff] == 1; ++axis) {}
```

Comment:
`std::find_if` ?

Reply:
I think, it's better that way because find_if would require creating a small function or a functor.

Reply:
done